src/main/java/org/prebid/server/events/EventUtil.java

```diff
@@ -39,15 +43,34 @@ public class EventUtil {
 
     private static final String LINE_ITEM_ID_PARAMETER = "l";
 
+    private static final Set<String> POSSIBLE_TYPES = stringValues(EventRequest.Type.class);
+    private static final Set<String> POSSIBLE_V_TYPES = stringValues(EventRequest.VastType.class);
+
     private EventUtil() {
     }
 
     public static void validateType(RoutingContext routingContext) {
         final String type = routingContext.request().params().get(TYPE_PARAMETER);
-        if (ObjectUtils.notEqual(type, IMP_TYPE) && ObjectUtils.notEqual(type, WIN_TYPE)) {
+        if (!POSSIBLE_TYPES.contains(type)) {
             throw new IllegalArgumentException(String.format(
-                    "Type '%s' is required query parameter. Possible values are %s and %s, but was %s",
-                    TYPE_PARAMETER, WIN_TYPE, IMP_TYPE, type));
+                    "Type '%s' is required query parameter. Possible values are %s, but was %s",
+                    TYPE_PARAMETER, POSSIBLE_TYPES, type));
+        }
+    }
+
+    public static void validateVType(RoutingContext routingContext) {
+        final String type = routingContext.request().params().get(TYPE_PARAMETER);
+        final String vType = routingContext.request().params().get(V_TYPE_PARAMETER);
+
+        if (type.equals(VAST_TYPE)) {
+            if (!POSSIBLE_V_TYPES.contains(vType)) {
+                throw new IllegalArgumentException(String.format(
+                        "Type '%s' is required query parameter. Possible values are %s, but was %s",
+                        V_TYPE_PARAMETER, POSSIBLE_V_TYPES, vType));
+            }
+        } else if (StringUtils.isNotBlank(vType)) {
+            throw new IllegalArgumentException(String.format("Type '%s' is only required for when '%s=%s'",
```

Comment:
`String.format("Parameter '%s' is only required for %s=%s", V_TYPE_PARAMETER, TYPE_PARAMETER, VAST_TYPE)`